data/serializer/src/main/java/tech/pegasys/teku/api/schema/AttesterSlashing.java

```diff
@@ -23,6 +26,14 @@ public AttesterSlashing(
     this.attestation_2 = new IndexedAttestation(attesterSlashing.getAttestation_2());
   }
 
+  @JsonCreator
+  public AttesterSlashing(
+      @JsonProperty("attestation_1") final IndexedAttestation attestation_1,
+      @JsonProperty("attestation_2") final IndexedAttestation attestation_2) {
+    this.attestation_1 = attestation_1;
+    this.attestation_2 = attestation_2;
+  }
+
```

Comment:
I'm not sure why this change was required as part of this PR.

Reply:
Ok I think I can see this one being useful for the submit block API which previously was a bug.  Still not sure what's deserializing an `IndexedAttestation` since they should only be in states and we shouldn't parse any of those from JSON (I think).

Reply:
We need to parse `IndexedAttestation` when we're deserializing JSON of `AttesterSlashing`, since it contains two `IndexedAttestation`s.

I needed to do deal with all this JSON stuff as I made sure now our `DataStructureUtil.randomBeaconBlock(int n)` method always returns a block with 1 `AttesterSlashing`, instead of returning an empty `AttesterSlashing` `SSZList` as it used to do. 

Then it turned out that we could not deserialize `BeaconBlock` JSON objects which included `AttesterSlashings`.

Reply:
oh right of course. Yep that makes sense.